### PR TITLE
Feature/combobox show add new value while autocompleting

### PR DESCRIPTION
**Pull request by it-vegard**
### Description

Enable option to add a new value while autocompleting (match searchText instead of value). 
Also highlight matches in FilteredOptions to better explain why the options is shown.

### Component Checklist 📝

- [ ] JSDoc
- [ ] Examples
- [ ] Documentation
- [ ] Storybook
- [ ] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [ ] Component tokens (`@navikt/core/css/tokens.json`)
- [ ] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [ ] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [ ] New component? CSS import (`@navikt/core/css/index.css`)
- [ ] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: .changeset/silver-rings-care.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@navikt/ds-react": patch
```
Comment (HalvorHaugan) [inline .changeset/silver-rings-care.md]: Should this be a `minor`?

**Comment by HalvorHaugan** [comment]
Side note: Haven't thought about this before, but doesn't the gray background of the "add" option look somewhat similar to the focus/hover state of the "normal" options? I understand that we want to distinguish it, though. I also think it's a bit weird that the focus/hover states are so different.
image

**Comment by KenAJoh** [comment]
> Side note: Haven't thought about this before, but doesn't the gray background of the "add" option look somewhat similar to the focus/hover state of the "normal" options? I understand that we want to distinguish it, though. I also think it's a bit weird that the focus/hover states are so different. image

🧠 Think it makes sense to just treat it like any other option visually

File: .changeset/silver-rings-care.md (hunk 1 above)
Comment (it-vegard) [reply .changeset/silver-rings-care.md]: Agreed. Changed it.

**Comment by it-vegard** [comment]
> > Side note: Haven't thought about this before, but doesn't the gray background of the "add" option look somewhat similar to the focus/hover state of the "normal" options? I understand that we want to distinguish it, though. I also think it's a bit weird that the focus/hover states are so different. image
> 
> 🧠 Think it makes sense to just treat it like any other option visually

Talked to Sjur about this, and he agrees that we should adjust this design. Sounds like maybe he talked to some of you about it as well? We agreed that it can happen outside of this PR, though.

**Comment by HalvorHaugan** [comment]
Hmm... I think you should take a look at chromatic.

File: @navikt/core/react/src/form/combobox/combobox.stories.tsx [hunk 2]
```diff
@@ -186,8 +185,6 @@ export const WithAddNewOptions: StoryFn = ({ open }: { open?: boolean }) => {
       options={options}
       allowNewValues={true}
       shouldAutocomplete={true}
-      value={value}
-      onChange={setValue}
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/combobox.stories.tsx]: I think removal of `value` here and in MultiSelectWithAddNewOptions "breaks" the Chromatic story.

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**